### PR TITLE
Changes to load secrets as part of environment configuration

### PR DESCRIPTION
**Pull request by vishwahiremat**
# Description

- Remove Secret Loader
- Adding changes to load secrets as part of LoadConfiguration
- Refactored the code accordingly

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #issue_number

### CODE REVIEW

File: pkg/corerp/datamodel/recipe_types.go [hunk 1]
```diff
@@ -52,7 +52,8 @@ type SecretConfig struct {
 	// The ID of an Applications.Core/SecretStore resource containing the Git platform personal access token (PAT). The secret
 	// store must have a secret named 'pat', containing the PAT value. A secret named
 	// 'username' is optional, containing the username associated with the pat. By default no username is specified.
-	Secret string `json:"secret,omitempty"`
+	Secret     string `json:"secret,omitempty"`
+	SecretData map[string]string
```
Comment (rynowak) [inline pkg/corerp/datamodel/recipe_types.go:56]: Why is the datamodel changing as part of this change? The data model is the internal representation of our API.

Comment (ytimocin) [reply pkg/corerp/datamodel/recipe_types.go:56]: Is Secret the Key and the SecretData the Value?

File: pkg/recipes/configloader/environment.go [hunk 2]
```diff
@@ -114,6 +114,17 @@ func getConfiguration(environment *v20231001preview.EnvironmentResource, applica
 
 	if environment.Properties.RecipeConfig != nil {
 		config.RecipeConfig = envDatamodel.Properties.RecipeConfig
+		secrets := config.RecipeConfig.Terraform.Authentication.Git.PAT
```
Comment (ytimocin) [inline pkg/recipes/configloader/environment.go:117]: This is specific to Terraform, right? Where do we check if this recipe is a Terraform recipe?

Comment (kachawla) [inline pkg/recipes/configloader/environment.go:117]: Ideally all the recipe type specific operations should be contained within driver, so configloader is not really the right place for this.

**Pull request closed without merging**